src/Controller/Api/DatasetSubmissionController.php

```diff
@@ -376,6 +376,48 @@ public function getFile(DatasetSubmission $datasetSubmission, Request $request):
         }
     }
 
+    /**
+     * Returns a list of file IDs.
+     *
+     * @param DatasetSubmission $datasetSubmission The id of the dataset submission.
+     * @param Request           $request           The request object.
+     *
+     * @Route(
+     *     "/api/file_dataset_submission_ids/{id}",
+     *     name="pelagos_api_get_file_ids_dataset_submission",
+     *     methods={"GET"},
+     *     defaults={"_format"="json"},
+     *     requirements={"id"="\d+"}
+     *     )
+     *
+     * @View()
+     *
+     * @return Response
+     */
+    public function getFileIds(DatasetSubmission $datasetSubmission, Request $request): Response
+    {
+        $this->denyAccessUnlessGranted('IS_AUTHENTICATED_FULLY');
```

Comment:
For API we could starting using the `@IsGranted` annotation, because a "fancy" screen isn't usually needed for API's.